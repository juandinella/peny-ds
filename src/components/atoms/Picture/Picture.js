import React from 'react'
import PropTypes from 'prop-types'
import styles from './Picture.styles'
import Skeleton from '../Skeleton'
import Utils from '../../../utils'

/**
 * Display image content.
 */

const Picture = ({
  id,
  src,
  minWidth,
  width,
  mWidth,
  height,
  altText,
  rounded,
  cover,
  fixedWidth,
  borderRadius,
  inline,
  className,
  shouldRender,
}) => {
  return (
    <Skeleton
      id={`${id}_skeleton`}
      show={!Utils.emptyValue(shouldRender) && !shouldRender()}
    >
      <picture
        id={id}
        data-testid={id}
        css={styles}
        inline={inline}
        className={className}
        rounded={rounded}
        borderRadius={borderRadius}
        cover={cover}
        mWidth={mWidth}
        width={width}
        minWidth={minWidth}
        height={height}
        fixedWidth={fixedWidth}
      >
        <img src={src} alt={altText} style={{ height }} />
      </picture>
    </Skeleton>
  )
}

Picture.displayName = 'Picture'

Picture.propTypes = {
  /** Picture's id */
  id: PropTypes.string,
  /** Source of the image */
  src: PropTypes.string.isRequired,
  /** Width of the image for mobile*/
  mWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Width of the image */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Min Width of the image */
  minWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Height of the image */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Mandatory text of the image */
  altText: PropTypes.string.isRequired,
  /** Adds border radius 100% */
  rounded: PropTypes.bool,
  /** Adds border radius 6px */
  borderRadius: PropTypes.bool,
  /** Inline-flex */
  inline: PropTypes.bool,
  /** Vertical align Middle */
  middle: PropTypes.bool,
  /** To add style */
  className: PropTypes.string,
  /** Adds border object-fit cover. This will center the image like background-size cover */
  cover: PropTypes.bool,
  /**
   * Function that decides if the component should be rendered.
   *
   * @return {boolean} false if component should not be rendered and show skeleton instead. Component renders otherwise.
   */
  shouldRender: PropTypes.func,
}

Picture.defaultProps = {
  height: 'auto',
}

export default Picture
