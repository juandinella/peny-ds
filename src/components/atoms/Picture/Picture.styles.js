import { css } from 'styled-components'
import layout from '../../../styles/theme/layout'

export default css`
  img {
    display: block;
    width: 100%;
  }

  ${(props) =>
    props.rounded &&
    css`
      img {
        border-radius: 100%;
      }
    `}

  ${(props) =>
    props.width &&
    css`
      img {
        max-width: ${(props) =>
          props.mWidth ? `${props.mWidth}px` : `${props.width}px`};
        @media screen and (min-width: ${() => layout.breakpoints.sm}) {
          max-width: ${(props) => `${props.width}px`};
        }
      }
    `}

  ${(props) =>
    props.minWidth &&
    css`
      img {
        min-width: ${(props) => `${props.minWidth}px`};
      }
    `}



  ${(props) =>
    props.borderRadius &&
    css`
      img {
        border-radius: var(--radius-sm);
      }
    `}

  ${(props) =>
    props.cover &&
    css`
      img {
        object-fit: cover;
      }
    `}

  ${(props) =>
    props.fixedWidth &&
    css`
      img {
        min-width: ${(props) => props.width};
      }
    `}

  ${(props) =>
    props.inline &&
    css`
      display: inline-flex;
    `}
`
