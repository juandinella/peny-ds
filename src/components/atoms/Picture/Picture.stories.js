import Picture from './Picture'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(Picture)

export default {
  title: 'Atoms/Picture',
  component: Picture,
  args: {
    src: 'https://picsum.photos/id/1033/400/400',
    width: 200,
    minWidth: 100,
    altText: 'A simple image',
  },
  argTypes: {
    width: { control: 'number' },
    height: { control: 'number' },
  },
}

export const Default = Template.bind({})

export const Height = Template.bind({})
Height.args = { src: 'https://picsum.photos/id/1033/400/800', height: 400 }

export const Width = Template.bind({})
Width.args = { src: 'https://picsum.photos/id/1033/800/400', width: 400 }

export const Skeleton = Template.bind({})
Skeleton.args = { shouldRender: () => false }
