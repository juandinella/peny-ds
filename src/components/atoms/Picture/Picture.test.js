import React from 'react'

import { render, cleanup } from '@testing-library/react'

import Picture from './Picture'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Atoms\/Picture$/,
})

const componentProps = {
  id: 'test-id',
  src: 'https://picsum.photos/id/1033/400/400',
  altText: 'Image description',
}

afterEach(cleanup)

describe('<Picture />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<Picture {...componentProps} />)

    const button = queryByTestId(componentProps.id)

    expect(button).toBeTruthy()
  })

  test('should render skeleton', () => {
    const { queryByTestId } = render(
      <Picture {...componentProps} shouldRender={() => false} />
    )

    expect(queryByTestId(`${componentProps.id}_skeleton`)).toBeTruthy()
  })
})
