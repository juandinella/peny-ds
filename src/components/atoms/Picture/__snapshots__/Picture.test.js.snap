// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Picture Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <picture
    className="sc-wzq26i-0 iKPofL"
    height="auto"
    width={200}
  >
    <img
      alt="A simple image"
      src="https://picsum.photos/400"
      style={
        Object {
          "height": "auto",
        }
      }
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Height 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <picture
    className="sc-wzq26i-0 iKPofL"
    height={400}
    width={200}
  >
    <img
      alt="A simple image"
      src="https://picsum.photos/id/1033/400/800"
      style={
        Object {
          "height": 400,
        }
      }
    />
  </picture>
</div>
`;

exports[`Storyshots Atoms/Picture Skeleton 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kjnynt-0 jfZhXD"
    data-testid="undefined_skeleton"
    id="undefined_skeleton"
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <picture
      className="sc-wzq26i-0 iKPofL"
      height="auto"
      width={200}
    >
      <img
        alt="A simple image"
        src="https://picsum.photos/400"
        style={
          Object {
            "height": "auto",
          }
        }
      />
    </picture>
  </div>
</div>
`;

exports[`Storyshots Atoms/Picture Width 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <picture
    className="sc-wzq26i-0 cRBjcr"
    height="auto"
    width={400}
  >
    <img
      alt="A simple image"
      src="https://picsum.photos/id/1033/800/400"
      style={
        Object {
          "height": "auto",
        }
      }
    />
  </picture>
</div>
`;
